### ISSUE TITLE
[OpenMP] UNREACHABLE executed at /root/llvm-project/llvm/tools/clang/lib/Basic/OpenMPKinds.cpp:855!

### ISSUE THREAD

**Issue by k-arrows** [llvm/llvm-project#139424]
Reproducer:
https://godbolt.org/z/nb864Yfxz
```cpp
void f(){
#pragma omp workshare
}
```

**Comment by AaronBallman** [comment]
The issue is an unhandled enumeration here: https://github.com/llvm/llvm-project/blob/2de0c927ae82eb0ae1d7b10028bf4704cb226cc0/clang/lib/Basic/OpenMPKinds.cpp#L796

Curiously, there are zero tests with `workshare` in `clang/test`. CC @bader

**Comment by alexey-bataev** [comment]
Workshare should be reported as an error, C/C++ do not support workshare construct

**Comment by alexey-bataev** [comment]
When flang added support for workshare, they did not report as unsupported in clang

**Comment by AaronBallman** [comment]
> When flang added support for workshare, they did not report as unsupported in clang

Oof, I don't know how we protect against those sort of issues, either. It would be a bummer to require Flang developers to add the functionality to Clang (or vice versa) whenever touching OpenMP, but that's a pretty high bar.

**Comment by AaronBallman** [comment]
CC'ing some Flang OpenMP folks: @tblah @kparzysz

**Comment by kparzysz** [comment]
Is it ok to emit `err_omp_unknown_directive`, or do we want a more detailed error message?

**Comment by AaronBallman** [comment]
> Is it ok to emit `err_omp_unknown_directive`, or do we want a more detailed error message?

I'm fine with an unknown directive diagnostic, but maybe @alexey-bataev has a different opinion.

**Comment by alexey-bataev** [comment]
I think that's the expected behaviour
